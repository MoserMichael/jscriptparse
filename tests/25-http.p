
cnt = 1

def startTxtSrv() {
    httpServer(9010, def(req,resp) {
        text = "
=====    
request url: {req.url()}  
method: {req.method()}    
headers {req.headers()}
user-agent: {req.header('User-Agent')}
"

        resp.send(200, "Pooh says hello. {cnt}" + text)
        cnt = cnt + 1
    })
}


def startTimeSrv() {
    httpServer(9010, def (req,resp) {
        println("url: {req.url()} url_: {req.url_}")
        if req.url() == "/time" {
            tm = localtime()
            js = toJsonString(tm)
            resp.send(200, js, "text/json")
        } else
            resp.send(501, "no one here")
    })
}

#startTimeSrv()

startTxtSrv()

httpSend('http://127.0.0.1:9010/abcd', none, def(resp,error) {
    println("response: {resp} error: {error}\n") 

    httpSend('http://127.0.0.1:9010/cdef', none, def(resp,error) {
        println("response: {resp} error: {error}\n") 
        exit(0)
    })
})

