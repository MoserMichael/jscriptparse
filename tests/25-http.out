status: 200 headers: none response: Pooh says hello. 1
=====    
request url: /abcd  
method: POST    
headers {"content-type":"text/json","content-length":"36","host":"127.0.0.1:9010","connection":"keep-alive"}
user-agent: none
requestData: { "name": "Pooh", "family": "Bear" }
 error: none

status: 200 headers: none response: Pooh says hello. 2
=====    
request url: /cdef  
method: GET    
headers {"host":"127.0.0.1:9010","connection":"keep-alive"}
user-agent: none
requestData: 
 error: none

